### ISSUE TITLE
Adding a distributed script and try to reproduce 86.6 mAP

### ISSUE THREAD

**Issue by SlongLiu** [Alibaba-MIIL/ASL#39]
Hi,

At first, thank you for sharing the training scripts. I really like your work which inspired me a lot. 

1)  There may be a bug in the `train.py` as mentioned in #38, which can be fixed by replacing line 155 `target = target` with `target = target.max(dim=1)[0]`.

2) I try to reproduce the results in the paper so that I add a distributed training scripts based on the `train.py` you shared, which can be found in [https://github.com/SlongLiu/ASL_reproduce/blob/master/train_dist.py](https://github.com/SlongLiu/ASL_reproduce/blob/master/train_dist.py). Based on the script, I can achieve the mAP 85.5 on COCO(448*448), which is lower than the result 86.6 mAP in paper. Could you please help me find out the reason of this gap?

Thanks for your help and happy new year!

**Comment by mrT23** [comment]
This is not the original code (i cannot release it due to commercial reasons), but a [reproduction ](https://github.com/Alibaba-MIIL/ASL/issues/36) code from the community.

Two differences i am aware of: In the original code we had better augmentations than randAugment and better EMA implementation. there could be other differences.

85.5 is not far from the article score. This result is significantly higher than previous SOTA, and achieved with a faster model and significantly less training time.
Try playing with the ema decay factor, learning rate schedule pct_start and number of epochs, and i predict you could easily surpass 86.0. we didn't do any major hyper-parameter optimization for the reproduction code

anyway, try training with ASL and compare to CE, and you will see the advantage of our proposed loss over the commonly used losses in multi-label

**Comment by SlongLiu** [comment]
Thanks for your reply and suggestions, that help me a lot. I will do more experiments and try other hyper parameters.

**Comment by mrT23** [comment]
@SlongLiu 
(1)
i updated the hyper-parameters (just lr and epoch adjustments)
now the run fully reproduces article results, 86.52, see attached file log:
[ltresnet_86.52.zip](https://github.com/Alibaba-MIIL/ASL/files/5941575/ltresnet_86.52.zip)

(2) i fixed the paths, although I am not sure that it will work on machines

**Comment by SlongLiu** [comment]
Thanks very much for your sharing and help!
